File: apps/web/src/content/curriculum.ts
/**
 * The 88-skill curriculum, client-side.
 *
 * ============================================================================
 * REVIEWED-BY:
 * ============================================================================
 * ^^ EMPTY. Every Arabic string below that is not a bare glyph is written by an
 * agent and has NOT been reviewed by a native Egyptian Arabic speaker. The
 * `code` and `labelAr` values are transcribed from
 * `services/api/seeds/curriculum.py` (itself transcribed from docs/02 §10.1);
 * `altAr` and the 25 unmarked letter keywords are new here and are
 * PLACEHOLDER. → REVIEW-QUEUE.md
 * ============================================================================
 *
 * ## Why this file exists at all
 *
 * The curriculum is owned by the API (C05 Content Service) and arrives in a
 * session manifest. This is not a second source of truth for it — it is the
 * *presentation* half the manifest does not and should not carry: which shape
 * to draw for `hh_spoon`, which hex fills `color_red`, how many counting dots
 * sit under `num_7`.
 *
 * `code` is the join key. When the API is wired up a manifest activity names
 * `skill_id: "color_red"` and the art registry finds the drawing by that code.
 * No artwork ships in the manifest and none needs to — that is what keeps a
 * session under the 4 MB budget in docs/06 §6.
 *
 * Codes and order match the seed exactly, so `introOrder` here is the same
 * `intro_order` the adaptive engine schedules against.
 */

export const REVIEWED_BY = "";

export type SkillCategory =
  | "colors"
  | "body_parts"
  | "social"
  | "household"
  | "numbers"
  | "letters";

/** The six categories in introduction order — most concrete first. */
export const CATEGORY_ORDER: readonly SkillCategory[] = [
  "colors",
  "body_parts",
  "social",
  "household",
  "numbers",
  "letters",
];

export interface Skill {
  /** Join key with the API. Never change one without changing the seed. */
  code: string;
  category: SkillCategory;
  /** What is written on screen. For letters and numbers this is the glyph. */
  labelAr: string;
  /** What Nour says. Egyptian colloquial; differs from labelAr for glyphs. */
  labelEgy: string;
  /** Image alt text. Mandatory — docs/06 §5 makes it a DB column for a reason. */
  altAr: string;
  /** 1–5, from the seed's `_tier_for`. Drives distractor contrast. */
  tier: number;
  introOrder: number;
  /** Colours only: the fill actually drawn, from the `--k-*` child palette. */
  hex?: string;
  /** Numbers only: how many counting dots to draw. */
  count?: number;
  /** Letters only: the keyword whose initial sound the letter makes. */
  keywordAr?: string;
}

/* -------------------------------------------------------------------------- */
/* Colours — 10. The hex values are the `--k-*` tokens from docs/06 §2, which   */
/* are the ones a child is taught to name. A "red" card filled with anything    */
/* other than --k-red teaches the wrong word for the wrong colour.              */
/* -------------------------------------------------------------------------- */

const COLOURS: Array<[suffix: string, label: string, hex: string, alt: string]> = [
  ["red", "أحمر", "#D93A2B", "دايرة حمرا"],
  ["blue", "أزرق", "#1C6BBF", "دايرة زرقا"],
  ["yellow", "أصفر", "#F0B429", "دايرة صفرا"],
  ["green", "أخضر", "#2E8B4A", "دايرة خضرا"],
  ["white", "أبيض", "#FFFFFF", "دايرة بيضا"],
  ["black", "أسود", "#2B2B2B", "دايرة سودا"],
  ["orange", "برتقالي", "#E87722", "دايرة برتقاني"],
  ["brown", "بني", "#8B5E3C", "دايرة بنّي"],
  ["pink", "وردي", "#D9538C", "دايرة وردي"],
  ["purple", "بنفسجي", "#7B4FA8", "دايرة بنفسجي"],
];

/* -------------------------------------------------------------------------- */
/* Body parts — 10                                                             */
/* -------------------------------------------------------------------------- */

const BODY_PARTS: Array<[suffix: string, label: string, alt: string]> = [
  ["head", "راس", "راس طفل"],
  ["hair", "شعر", "شعر"],
  ["eye", "عين", "عين"],
  ["ear", "ودن", "ودن"],
  ["nose", "مناخير", "مناخير"],
  ["mouth", "بق", "بق بيبتسم"],
  ["teeth", "سنان", "سنان"],
  ["hand", "إيد", "إيد مفتوحة"],
  ["leg", "رجل", "رجل"],
  ["tummy", "بطن", "بطن"],
];

/* -------------------------------------------------------------------------- */
/* Social phrases — 10. Drawn as a moment, not as an object: "صباح الخير" is a  */
/* sun over a house, because there is no picture of a greeting.                 */
/* -------------------------------------------------------------------------- */

const SOCIAL: Array<[suffix: string, label: string, alt: string]> = [
  ["good_morning", "صباح الخير", "شمس طالعة الصبح"],
  ["goodbye", "مع السلامة", "إيد بتسلّم"],
  ["thanks", "شكرا", "قلب في الإيد"],
  ["please", "من فضلك", "إيد مفتوحة بتطلب"],
  ["yes", "أيوه", "علامة صح"],
  ["no", "لأ", "إيد بتقول لأ"],
  ["come", "تعالى", "إيد بتنادي"],
  ["dad", "بابا", "وش بابا"],
  ["mum", "ماما", "وش ماما"],
  ["my_name", "اسمي", "طفل بيشاور على نفسه"],
];

/* -------------------------------------------------------------------------- */
/* Household objects — 20                                                      */
/* -------------------------------------------------------------------------- */

const HOUSEHOLD: Array<[suffix: string, label: string, alt: string]> = [
  ["toothbrush", "فرشة سنان", "فرشة سنان"],
  ["toothpaste", "معجون", "أنبوبة معجون"],
  ["soap", "صابونة", "صابونة"],
  ["towel", "فوطة", "فوطة متعلقة"],
  ["fork", "شوكة", "شوكة"],
  ["spoon", "معلقة", "معلقة"],
  ["knife", "سكينة", "سكينة"],
  ["plate", "طبق", "طبق"],
  ["cup", "كوباية", "كوباية"],
  ["water_glass", "كوباية مية", "كوباية فيها مية"],
  ["chair", "كرسي", "كرسي"],
  ["table", "ترابيزة", "ترابيزة"],
  ["bed", "سرير", "سرير"],
  ["pillow", "مخدة", "مخدة"],
  ["door", "باب", "باب"],
  ["window", "شباك", "شباك"],
  ["shoes", "جزمة", "جزمة"],
  ["bag", "شنطة", "شنطة"],
  ["clothes", "لبس", "تي شيرت"],
  ["comb", "مشط", "مشط"],
];

/* -------------------------------------------------------------------------- */
/* Numbers 1–10. `labelAr` is the Eastern Arabic glyph — docs/06 §1 requires    */
/* Eastern Arabic numerals for everything a child sees.                        */
/* -------------------------------------------------------------------------- */

const NUMERALS: Array<[value: number, glyph: string, word: string]> = [
  [1, "١", "واحد"],
  [2, "٢", "اتنين"],
  [3, "٣", "تلاتة"],
  [4, "٤", "أربعة"],
  [5, "٥", "خمسة"],
  [6, "٦", "ستة"],
  [7, "٧", "سبعة"],
  [8, "٨", "تمانية"],
  [9, "٩", "تسعة"],
  [10, "١٠", "عشرة"],
];

/* -------------------------------------------------------------------------- */
/* Letters — the 28 of the Arabic alphabet.                                    */
/*                                                                             */
/* `nameAr` is what the letter is CALLED, and is what Nour says. `keywordAr`  */
/* is the word whose initial sound it makes. docs/02 §10.1                     */
/* gives three (أ → أسد, ب → بطة, ت → تفاحة) and those are transcribed. The     */
/* other 25 are PLACEHOLDER: chosen for concreteness and for being words a      */
/* child in Egypt already hears at home, but a phonics decision belongs to a    */
/* speech therapist, not to the agent that wrote this list.                     */
/* -------------------------------------------------------------------------- */

const LETTERS: Array<[name: string, glyph: string, nameAr: string, keyword: string]> = [
  ["alef", "أ", "ألف", "أسد"],
  ["baa", "ب", "با", "بطة"],
  ["taa", "ت", "تا", "تفاحة"],
  ["thaa", "ث", "ثا", "ثعلب"],
  ["jeem", "ج", "جيم", "جمل"],
  ["haa", "ح", "حا", "حصان"],
  ["khaa", "خ", "خا", "خروف"],
  ["dal", "د", "دال", "ديك"],
  ["thal", "ذ", "ذال", "ذرة"],
  ["raa", "ر", "را", "رمانة"],
  ["zay", "ز", "زاي", "زرافة"],
  ["seen", "س", "سين", "سمكة"],
  ["sheen", "ش", "شين", "شمس"],
  ["sad", "ص", "صاد", "صاروخ"],
  ["dad", "ض", "ضاد", "ضفدع"],
  ["tah", "ط", "طا", "طيارة"],
  ["zah", "ظ", "ظا", "ظرف"],
  ["ain", "ع", "عين", "عصفورة"],
  ["ghain", "غ", "غين", "غزالة"],
  ["faa", "ف", "فا", "فيل"],
  ["qaf", "ق", "قاف", "قمر"],
  ["kaf", "ك", "كاف", "كتاب"],
  ["lam", "ل", "لام", "ليمونة"],
  ["meem", "م", "ميم", "موزة"],
  ["noon", "ن", "نون", "نحلة"],
  ["haa2", "ه", "ها", "هدية"],
  ["waw", "و", "واو", "وردة"],
  ["yaa", "ي", "يا", "يد"],
];

/** The three keywords docs/02 §10.1 actually supplies. The rest need review. */
export const REVIEWED_LETTER_KEYWORDS: readonly string[] = ["alef", "baa", "taa"];

/** Mirrors `_tier_for` in services/api/seeds/curriculum.py. */
function tierFor(category: SkillCategory, index: number): number {
  if (category === "colors" || category === "body_parts") return index < 5 ? 1 : 2;
  if (category === "household") return index < 10 ? 2 : 3;
  if (category === "social") return index < 4 ? 1 : 2;
  if (category === "numbers") return index < 5 ? 2 : 3;
  return index < 14 ? 3 : 4; // letters
}

function build(): Skill[] {
  const skills: Skill[] = [];
  let order = 0;

  const add = (item: Omit<Skill, "introOrder">) => {
    skills.push({ ...item, introOrder: order });
    order += 1;
  };

  COLOURS.forEach(([suffix, label, hex, alt], index) =>
    add({
      code: `color_${suffix}`,
      category: "colors",
      labelAr: label,
      labelEgy: label,
      altAr: alt,
      tier: tierFor("colors", index),
      hex,
    }),
  );

  BODY_PARTS.forEach(([suffix, label, alt], index) =>
    add({
      code: `body_${suffix}`,
      category: "body_parts",
      labelAr: label,
      labelEgy: label,
      altAr: alt,
      tier: tierFor("body_parts", index),
    }),
  );

  SOCIAL.forEach(([suffix, label, alt], index) =>
    add({
      code: `social_${suffix}`,
      category: "social",
      labelAr: label,
      labelEgy: label,
      altAr: alt,
      tier: tierFor("social", index),
    }),
  );

  HOUSEHOLD.forEach(([suffix, label, alt], index) =>
    add({
      code: `hh_${suffix}`,
      category: "household",
      labelAr: label,
      labelEgy: label,
      altAr: alt,
      tier: tierFor("household", index),
    }),
  );

  NUMERALS.forEach(([value, glyph, word], index) =>
    add({
      code: `num_${value}`,
      category: "numbers",
      labelAr: glyph,
      labelEgy: word,
      altAr: `${word} — ${glyph}`,
      tier: tierFor("numbers", index),
      count: value,
    }),
  );

  LETTERS.forEach(([name, glyph, nameAr, keyword], index) =>
    add({
      code: `letter_${name}`,
      category: "letters",
      labelAr: glyph,
      // The letter's NAME, not its keyword.
      //
      // `label_egy` in the API seed is the keyword — `أ` carries `أسد` — and
      // taking that literally produced an instruction reading "وريني زرافة"
      // ("show me a giraffe") beside a card showing `ز` and a card showing `٥`.
      // There was no giraffe on the screen. What Nour says has to name the
      // thing the child is looking at, so a letter is spoken as a letter, and
      // the keyword appears only in the story moment, where it is introduced
      // as one ("أ زي أسد"). The seed has the same latent bug for TTS.
      // → REVIEW-QUEUE.md #14
      labelEgy: nameAr,
      altAr: `حرف ${glyph}`,
      tier: tierFor("letters", index),
      keywordAr: keyword,
    }),
  );

  return skills;
}

export const SKILLS: readonly Skill[] = build();

const BY_CODE = new Map(SKILLS.map((item) => [item.code, item]));

export function skill(code: string): Skill | undefined {
  return BY_CODE.get(code);
}

export function skillsIn(category: SkillCategory): Skill[] {
  return SKILLS.filter((item) => item.category === category);
}

/**
 * The label a child hears, which is not always the label they see.
 *
 * `٣` is written but "تلاتة" is said, and an instruction reading "وريني ٣"
 * would be a sentence no adult says out loud to a child.
 */
export function spokenLabel(item: Skill): string {
  return item.labelEgy;
}

File: services/api/seeds/curriculum.py
"""The 88-skill curriculum.

================================================================================
REVIEWED-BY:
================================================================================
^^ That header is EMPTY and CI fails while it stays empty (P06 requires it).
It must carry the name of a native Egyptian Arabic speaker who has reviewed
every label in this file.

What is transcribed vs. what is invented:

  TRANSCRIBED from docs/02-data-model.md §10.1 — the architecture package is
  authoritative for these, and they are reproduced verbatim:
      * the 10 colour words        * the 10 body parts
      * the 20 household objects   * the 10 social phrases
      * the 28 letters and the ten numerals

  PLACEHOLDER — generated mechanically, NOT reviewed, NOT correct:
      * label_vowelised   (tashkeel) — drives TTS. Wrong tashkeel means Nour
                          mispronounces a word to a child learning to speak.
      * phonemes          (IPA) — drives the pronunciation similarity scorer.
      * transliteration
      * label_egy         where it differs from MSA
      * every distractor pool

A wrong vowelisation is not cosmetic here. The whole product is a child hearing
a word said correctly, repeatedly. → REVIEW-QUEUE.md #6
================================================================================
"""

from __future__ import annotations

from dataclasses import dataclass, field

#: Set to a reviewer's name to satisfy the CI check. Must not be a placeholder.
REVIEWED_BY: str = ""

PLACEHOLDER = "PLACEHOLDER-NOT-REVIEWED"

MIN_DISTRACTORS = 4


@dataclass(frozen=True, slots=True)
class Skill:
    code: str
    category: str
    label_ar: str
    label_vowelised: str
    label_egy: str
    transliteration: str
    phonemes: str
    difficulty_tier: int
    intro_order: int
    alt_text_ar: str
    #: Perceptual colour, as (L, a, b)-ish coordinates normalised to 0-1, used
    #: by the tier-1 contrast rule. None for non-colour skills.
    colour: tuple[float, float, float] | None = None
    distractor_pool: tuple[str, ...] = field(default_factory=tuple)
    prerequisites: tuple[str, ...] = field(default_factory=tuple)


# --- transcribed from docs/02 §10.1 ----------------------------------------

LETTERS: tuple[tuple[str, str], ...] = (
    ("alef", "أ"),
    ("baa", "ب"),
    ("taa", "ت"),
    ("thaa", "ث"),
    ("jeem", "ج"),
    ("haa", "ح"),
    ("khaa", "خ"),
    ("dal", "د"),
    ("thal", "ذ"),
    ("raa", "ر"),
    ("zay", "ز"),
    ("seen", "س"),
    ("sheen", "ش"),
    ("sad", "ص"),
    ("dad", "ض"),
    ("tah", "ط"),
    ("zah", "ظ"),
    ("ain", "ع"),
    ("ghain", "غ"),
    ("faa", "ف"),
    ("qaf", "ق"),
    ("kaf", "ك"),
    ("lam", "ل"),
    ("meem", "م"),
    ("noon", "ن"),
    ("haa2", "ه"),
    ("waw", "و"),
    ("yaa", "ي"),
)

#: Keyword words, per docs/02 §10.1 ("أ → أَسَد, ب → بَطَّة, ت → تُفَّاحة …").
#: Only the three the document actually gives are real; the rest are
#: PLACEHOLDERS and are marked as such.
LETTER_KEYWORDS: dict[str, str] = {
    "alef": "أسد",
    "baa": "بطة",
    "taa": "تفاحة",
}

NUMERALS: tuple[tuple[str, str, str], ...] = (
    ("1", "١", "واحد"),
    ("2", "٢", "اتنين"),
    ("3", "٣", "تلاتة"),
    ("4", "٤", "أربعة"),
    ("5", "٥", "خمسة"),
    ("6", "٦", "ستة"),
    ("7", "٧", "سبعة"),
    ("8", "٨", "تمانية"),
    ("9", "٩", "تسعة"),
    ("10", "١٠", "عشرة"),
)

#: (code suffix, label, approximate perceptual coordinates).
#: The coordinates are a PLACEHOLDER stand-in for real CIELAB values; they exist
#: so the tier-1 contrast rule has something to compute on.
COLOURS: tuple[tuple[str, str, tuple[float, float, float]], ...] = (
    ("red", "أحمر", (0.53, 0.90, 0.75)),
    ("blue", "أزرق", (0.42, 0.30, 0.10)),
    ("yellow", "أصفر", (0.86, 0.55, 0.95)),
    ("green", "أخضر", (0.55, 0.15, 0.75)),
    ("white", "أبيض", (1.00, 0.50, 0.50)),
    ("black", "أسود", (0.00, 0.50, 0.50)),
    ("orange", "برتقالي", (0.65, 0.78, 0.88)),
    ("brown", "بني", (0.40, 0.62, 0.68)),
    ("pink", "وردي", (0.75, 0.80, 0.55)),
    ("purple", "بنفسجي", (0.45, 0.68, 0.25)),
)

BODY_PARTS: tuple[tuple[str, str], ...] = (
    ("head", "راس"),
    ("hair", "شعر"),
    ("eye", "عين"),
    ("ear", "ودن"),
    ("nose", "مناخير"),
    ("mouth", "بق"),
    ("teeth", "سنان"),
    ("hand", "إيد"),
    ("leg", "رجل"),
    ("tummy", "بطن"),
)

HOUSEHOLD: tuple[tuple[str, str], ...] = (
    ("toothbrush", "فرشة سنان"),
    ("toothpaste", "معجون"),
    ("soap", "صابونة"),
    ("towel", "فوطة"),
    ("fork", "شوكة"),
    ("spoon", "معلقة"),
    ("knife", "سكينة"),
    ("plate", "طبق"),
    ("cup", "كوباية"),
    ("water_glass", "كوباية مية"),
    ("chair", "كرسي"),
    ("table", "ترابيزة"),
    ("bed", "سرير"),
    ("pillow", "مخدة"),
    ("door", "باب"),
    ("window", "شباك"),
    ("shoes", "جزمة"),
    ("bag", "شنطة"),
    ("clothes", "لبس"),
    ("comb", "مشط"),
)

SOCIAL: tuple[tuple[str, str], ...] = (
    ("good_morning", "صباح الخير"),
    ("goodbye", "مع السلامة"),
    ("thanks", "شكرا"),
    ("please", "من فضلك"),
    ("yes", "أيوه"),
    ("no", "لأ"),
    ("come", "تعالى"),
    ("dad", "بابا"),
    ("mum", "ماما"),
    ("my_name", "اسمي"),
)


def _placeholder_vowelised(label: str) -> str:
    """A stand-in for real tashkeel.

    Deliberately returns the bare label rather than guessing diacritics: a wrong
    vowelisation would be silently *worse* than none, because it would be
    plausible enough to ship and would make Nour mispronounce the word.
    """
    return label


def _placeholder_phonemes(code: str) -> str:
    return f"{PLACEHOLDER}:{code}"


def _tier_for(category: str, index: int) -> int:
    """Introduction difficulty, 1-5. A mechanical placeholder ordering."""
    if category in ("colors", "body_parts"):
        return 1 if index < 5 else 2
    if category == "household":
        return 2 if index < 10 else 3
    if category == "social":
        return 1 if index < 4 else 2
    if category == "numbers":
        return 2 if index < 5 else 3
    return 3 if index < 14 else 4  # letters


def build_skills() -> list[Skill]:
    """All 88 skills, in introduction order. Deterministic."""
    skills: list[Skill] = []
    order = 0

    def add(
        code: str,
        category: str,
        label: str,
        *,
        index: int,
        colour: tuple[float, float, float] | None = None,
        egy: str | None = None,
    ) -> None:
        nonlocal order
        skills.append(
            Skill(
                code=code,
                category=category,
                label_ar=label,
                label_vowelised=_placeholder_vowelised(label),
                label_egy=egy or label,
                transliteration=f"{PLACEHOLDER}:{code}",
                phonemes=_placeholder_phonemes(code),
                difficulty_tier=_tier_for(category, index),
                intro_order=order,
                alt_text_ar=f"[{PLACEHOLDER}] صورة {label}",
                colour=colour,
            )
        )
        order += 1

    # Order matters: the earliest categories are the most concrete and the most
    # present in a child's daily life, which is where errorless learning starts.
    for index, (suffix, label, coordinates) in enumerate(COLOURS):
        add(f"color_{suffix}", "colors", label, index=index, colour=coordinates)
    for index, (suffix, label) in enumerate(BODY_PARTS):
        add(f"body_{suffix}", "body_parts", label, index=index)
    for index, (suffix, label) in enumerate(SOCIAL):
        add(f"social_{suffix}", "social", label, index=index)
    for index, (suffix, label) in enumerate(HOUSEHOLD):
        add(f"hh_{suffix}", "household", label, index=index)
    for index, (suffix, glyph, word) in enumerate(NUMERALS):
        add(f"num_{suffix}", "numbers", glyph, index=index, egy=word)
    for index, (name, glyph) in enumerate(LETTERS):
        keyword = LETTER_KEYWORDS.get(name)
        add(
            f"letter_{name}",
            "letters",
            glyph,
            index=index,
            # A SINGLE token: the real keyword words are one word each
            # (أ → أسد), and a multi-word placeholder would silently push
            # every instruction over the 5-word limit and mask the real
            # constraint. Still visibly a placeholder.
            egy=keyword or f"{PLACEHOLDER}-{glyph}",
        )

    return _attach_distractor_pools(skills)


def _attach_distractor_pools(skills: list[Skill]) -> list[Skill]:
    """Give every skill at least MIN_DISTRACTORS candidates.

    PLACEHOLDER. docs/04c §C05 requires these to be HAND-CURATED so that wrong
    answers are perceptually and semantically distinct — "a red card must never
    sit next to an orange card at tier 1". This function cannot do that: it
    simply offers every skill outside the target's own category, and leaves the
    real discrimination work to `choose_distractors`, which applies the
    contrast rules at selection time.

    A curated pool is still required before launch. → REVIEW-QUEUE.md #6
    """
    by_category: dict[str, list[str]] = {}
    for skill in skills:
        by_category.setdefault(skill.category, []).append(skill.code)

    resolved: list[Skill] = []
    for skill in skills:
        pool = [
            code
            for category, codes in by_category.items()
            if category != skill.category
            for code in codes
        ]
        # Deterministic spread rather than a random sample, so the seed is
        # reproducible and a diff is readable.
        stride = max(1, len(pool) // 12)
        selected = tuple(pool[::stride][:12])
        resolved.append(
            Skill(
                code=skill.code,
                category=skill.category,
                label_ar=skill.label_ar,
                label_vowelised=skill.label_vowelised,
                label_egy=skill.label_egy,
                transliteration=skill.transliteration,
                phonemes=skill.phonemes,
                difficulty_tier=skill.difficulty_tier,
                intro_order=skill.intro_order,
                alt_text_ar=skill.alt_text_ar,
                colour=skill.colour,
                distractor_pool=selected,
                prerequisites=skill.prerequisites,
            )
        )
    return resolved


# --- activity templates, docs/02 §10.2 -------------------------------------


@dataclass(frozen=True, slots=True)
class ActivityTemplate:
    code: str
    kind: str
    modality: str
    choice_count: int
    #: MUST be <= 5 words after {label} substitution (docs/06 §4).
    instruction_ar: str
    success_audio_pool: tuple[str, ...]
    retry_audio_pool: tuple[str, ...]
    min_tier: int


#: PLACEHOLDER Arabic. Every string here is spoken to a child and must be
#: reviewed by a native speaker. The success lines are warm; the retry lines are
#: gentle and NEVER corrective — there is no failure state in this product.
SUCCESS_POOL: tuple[str, ...] = ("برافو!", "شاطر أوي!", "أحسنت!", "تمام!")
RETRY_POOL: tuple[str, ...] = ("يلا نجرب تاني", "خد وقتك", "تعالى نشوف")

ACTIVITY_TEMPLATES: tuple[ActivityTemplate, ...] = (
    ActivityTemplate(
        "listen_point_2choice",
        "listen_point",
        "receptive",
        2,
        "وريني {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        1,
    ),
    ActivityTemplate(
        "listen_point_3choice",
        "listen_point",
        "receptive",
        3,
        "وريني {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        2,
    ),
    ActivityTemplate(
        "listen_point_4choice",
        "listen_point",
        "receptive",
        4,
        "وريني {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        3,
    ),
    ActivityTemplate(
        "match_pair_picture",
        "match_pair",
        "receptive",
        2,
        "هات زي {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        2,
    ),
    ActivityTemplate(
        "match_pair_word",
        "match_pair",
        "receptive",
        2,
        "وصّل {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        3,
    ),
    ActivityTemplate(
        "say_it_word",
        "say_it",
        "expressive",
        1,
        "قول {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        2,
    ),
    ActivityTemplate(
        "say_it_repeat_after",
        "say_it",
        "expressive",
        1,
        "قول ورايا {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        1,
    ),
    ActivityTemplate(
        "sort_category_2bin",
        "sort_category",
        "receptive",
        2,
        "حط {label} هنا",
        SUCCESS_POOL,
        RETRY_POOL,
        3,
    ),
    ActivityTemplate(
        "story_moment_3frame",
        "story_moment",
        "receptive",
        3,
        "فين {label}",
        SUCCESS_POOL,
        RETRY_POOL,
        2,
    ),
)
